Source/Dafny/Rewriter.cs

```diff
@@ -115,14 +118,69 @@ public abstract class IRewriter {
       Contract.Requires(program != null);
     }
   }
-
+  /*
+   * Code Example that displays the Warnings
+   *
+   * datatype Color = Red | Green | ShadesOfGray(nat)
+      method MonochromaticMethod(c: Color) returns (x: bool) {
+      return match c
+            case Green => true
+                 ^^^ Warning: Constructor name 'Green' should contain a parenthesis at the end           
+            case anythingElse => false;
+        }
+        
+    * datatype Color = Red | Green | ShadesOfGray(nat)
+      method MonochromaticMethod(c: Color) returns (x: bool) {
+      return match c
+            case ShadesOfGray => true
+                  ^^^ Warning: Variable name 'ShadesOfGray' on line 129 starts with a capital letter
+            case anythingElse => false;
+    }
+   */
+  class ConstructorWarning : IRewriter {
+    internal override void PostResolve(Program program) {
+      base.PostResolve(program);
+      foreach (var moduleDefinition in program.Modules()) {
+        foreach (var topLevelDecl in moduleDefinition.TopLevelDecls.OfType<TopLevelDeclWithMembers>()) {
+          foreach (var callable in topLevelDecl.Members.OfType<ICallable>()) {
+            var visitor = new ConstructorWarningVisitor(this.Reporter);
+            visitor.Visit(callable, Unit.Default);
+          }
+        }
+      }
+    }
+    public ConstructorWarning(ErrorReporter reporter) : base(reporter) {
+    }
+  }
+  class ConstructorWarningVisitor : TopDownVisitor<Unit> {
+    private readonly ErrorReporter reporter;
+    public ConstructorWarningVisitor(ErrorReporter reporter) {
+      this.reporter = reporter;
+    }
+    // Implements warning for constructors in match which ensures constructor name has parenthesis at the end and Begins with an uppercase letter. 
+    protected override bool VisitOneExpr(Expression expr, ref Unit st) {
+      if (expr is NestedMatchExpr matchExpr) {
+        var matchExprCases = matchExpr.Cases;
+        foreach (var caseExpr in matchExprCases) {
+          if (caseExpr.Pat is IdPattern idPattern) {
+            var isConstructor = idPattern.Arguments != null;
+            var isVariable = idPattern.Arguments == null;
+            if (idPattern.HasParenthesis == false && isConstructor) {
```

Comment:
`idPattern.HasParenthesis == false` can be rewritten to `!idPattern.HasParenthesis`